src/tests/math_tests.rs

```diff
@@ -0,0 +1,77 @@
+use widestring::utf32str;
+
+use super::super::prelude::*;
+
+use crate::{
+    builtins::{math, prelude::Parser},
+    io::{IoChain, IoStreams, OutputStream, StringOutputStream},
+};
+
+fn test_math(
+    argv: &mut [&wstr],
+    expected_ans: &str,
+    parser: &&Parser,
+    stream: &mut IoStreams,
+    ans_lst: &mut String,
+) {
+    let _res = math::math(parser, stream, argv);
+    ans_lst.push_str(expected_ans);
+    ans_lst.push_str("\n");
+    assert!(stream.out.contents() == ans_lst);
+}
+
+#[test]
+fn test_scale_mode() {
+    let _cleanup = test_init();
+    let parser = &Parser::principal_parser();
+    let mut out = OutputStream::String(StringOutputStream::new());
+    let mut err = OutputStream::String(StringOutputStream::new());
+    let io_chain = IoChain::new();
+    let mut stream = IoStreams::new(&mut out, &mut err, &io_chain);
+    let mut ans_lst = String::new();
+
+    let mut argv = [
+        utf32str!("math"),
```

Comment:
I would usually add these tests as scripted tests in https://github.com/fish-shell/fish-shell/blob/46f6aa802489342b73782f61d7a669349b4874a3/tests/checks/math.fish instead.

That's more easily readable, because it is actually used in a script, and can skip all this setup gunk.

Cargo tests are fine for something that's not directly exposed in scripting, but this is, so that's the nicer place to test it.